src/bin/pg_autoctl/cli_service.c

```diff
@@ -238,6 +211,50 @@ cli_monitor_run(int argc, char **argv)
 }
 
 
+/*
+ * ensure_monitor_running checks if monitor is running, attempts to restart it if it is not.
+ * verifies the extension version is the same as expected and returns the postgres uri for the
+ * monitor connection upon success.
+ */
+static void
+ensure_monitor_running(Monitor *monitor, MonitorConfig *mconfig)
+{
+	MonitorExtensionVersion version = { 0 };
+	char postgresUri[MAXCONNINFO];
+
+	if (!pg_is_running(mconfig->pgSetup.pg_ctl, mconfig->pgSetup.pgdata))
+	{
+		log_info("Postgres is not running, starting postgres");
+
+		if (!pg_ctl_start(mconfig->pgSetup.pg_ctl,
+						  mconfig->pgSetup.pgdata,
+						  mconfig->pgSetup.pgport,
+						  mconfig->pgSetup.listen_addresses))
+		{
+			log_error("Failed to start PostgreSQL, see above for details");
+			exit(EXIT_CODE_PGCTL);
+		}
+	}
+
+	log_info("PostgreSQL is running in \"%s\" on port %d",
+			 mconfig->pgSetup.pgdata, mconfig->pgSetup.pgport);
```

Comment:
So this INFO line is getting printed at every loop iteration on the monitor output then? That would be too much chatter, let's only print that when we restart Postgres. It might be that reusing `ensure_local_postgres_is_running` would give that for us for free, though, please have a look if we can re-use that on the monitor side of things.